### PR TITLE
MAINT: interpolate: csr_matrix -> csr_array

### PR DESCRIPTION
**Pull request by ev-br**
Return `csr_array` objects instead of `csr_matrix` from `BSpline.design_marix` method.

The main question is if this should go to 1.8.x or 1.9.x : the method is new in 1.8.0, so it makes sense to tweak the return type right away to avoid even thinking about backcompat down the line (however mild it will be). 
On the other hand, this is directly user-facing, and AFAIU the idea is to try the `_array` classes first before making a general recommendation. 
So I'm deferring the decision to @tylerjereddy and @rgommers .

### CODE REVIEW

**Comment by ev-br** [comment]
CI failures seem unrelated

**Comment by rgommers** [comment]
This is a tiny patch, so should be okay to backport. Changing it seems fine to me. We've had two RCs though, so not sure @tylerjereddy is happy with a design change at this point. I'd say he has the final say on this one.

**Review by tylerjereddy — APPROVED** [review]
Ok, merging with backport label based on core dev recommendations.

I believe we will need *at least* an rc3 anyway b/c of PROPACK challenges.

**Pull request merged**
